src/test/java/ro/developmentfactory/thedrones/controller/DronePositionControllerIntegrationTest.java

```diff
@@ -0,0 +1,126 @@
+        UUID droneId = insertDefaultDroneAndDroneStatus();
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/drones/{id}/turn/{direction}", droneId, turnDirection.name())
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.facingDirection").value("W"))
+                .andReturn();
+    }
+
+    @Test
+    @DisplayName("Test move forward successfully")
+    public void testMoveForward() throws Exception {
+        UUID droneId = insertDefaultDroneAndDroneStatus();
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/drones/{id}/goForward", droneId)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.currentPositionX").value(DRONE_STATUS_POSITION_X ))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.currentPositionY").value(DRONE_STATUS_POSITION_Y + 1))
+                .andReturn();
+    }
+
+    @Test
+    @DisplayName("Test move forward with non-existing drone")
+    public void testMoveForwardNonExistingDrone() throws Exception {
+        UUID nonExistingDroneId = UUID.randomUUID();
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/drones/{id}/goForward", nonExistingDroneId)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isNotFound())
+                .andReturn();
+    }
+
+    @Test
+    @DisplayName("Test turn multiple times")
+    public void testTurnMultipleTimes() throws Exception {
+        UUID droneId = insertDefaultDroneAndDroneStatus();
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/drones/{id}/turn/{direction}", droneId, TurnDirection.RIGHT.name())
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isOk());
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/drones/{id}/turn/{direction}", droneId, TurnDirection.RIGHT.name())
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.facingDirection").value("S"))
+                .andReturn();
+    }
+
+    @Test
+    @DisplayName("Test move forward at the edge of the field")
+    public void testMoveForwardAtEdge() throws Exception {
+        UUID droneId = insertDefaultDroneAndDroneStatusAtEdge();
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/drones/{id}/goForward", droneId)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isBadRequest())
+                .andReturn();
+    }
+
+    private UUID insertDefaultDroneAndDroneStatusAtEdge() {
```

Comment:
this method is very similar to `IntegrationTest.insertDefaultDroneAndDroneStatus()`. it would be great to create a way to generate your test data with a Factory or a Prototype pattern and then pass it for saving